### PR TITLE
Search for declined (BRIDGE-3303)

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java [hunk 1]
```diff
@@ -70,7 +70,9 @@ protected QueryBuilder createQuery(AdherenceRecordsSearch search) {
         builder.append(BASE_QUERY);
         
         WhereClauseBuilder where = builder.startWhere(AND);
-        where.appendRequired("ar.userId = :userId", "userId", search.getUserId());
+        if (search.getUserId() != null) {
```
Comment (jnelsonjava) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java:73]: Might be worth adding a test in HibernateAdherenceRecordDaoTest for a null userId.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java [hunk 2]
```diff
@@ -91,6 +93,8 @@ protected QueryBuilder createQuery(AdherenceRecordsSearch search) {
             where.append("tm.timeWindowGuid IN :timeWindowGuids",
                     "timeWindowGuids", search.getTimeWindowGuids());
         }
+        where.appendBoolean("declined", search.isDeclined());
```
Comment (jnelsonjava) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java:96]: This could use a test as well.

**Review by jnelsonjava — APPROVED** [review]
Approved. Could add unit tests referenced in comments. But I see that the integration tests seem to cover that as well.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java:73]: Added

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAdherenceRecordDao.java:96]: Added
